sound/soc/sof/ipc4-topology.c

```diff
@@ -1387,6 +1469,85 @@ static int sof_ipc4_prepare_src_module(struct snd_sof_widget *swidget,
 	return 0;
 }
 
+static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
+					   struct snd_pcm_hw_params *fe_params,
+					   struct snd_sof_platform_stream_params *platform_params,
+					   struct snd_pcm_hw_params *pipeline_params, int dir)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+	struct snd_soc_dapm_widget *widget = swidget->widget;
+	struct sof_ipc4_process *process = swidget->private;
+	struct sof_ipc4_available_audio_format *available_fmt = &process->available_fmt;
+	struct sof_ipc4_control_data *control_data;
+	struct snd_sof_control *scontrol = NULL;
+	void *cfg = process->ipc_config_data;
+	const struct snd_kcontrol_new *kc;
+	struct soc_bytes_ext *sbe;
+	void *extra_flag;
+	int ret, i;
+
+	available_fmt->ref_audio_fmt = &available_fmt->base_config->audio_fmt;
```

Comment:
I think we can but all other modules do it in the prepare callback function. If we want to move the set operation in the setup time, I think we need a separate patch. I guess we do it in the prepare function because we need to use it only in this function.